File: final_project/backend/db/crud/categories.go
package crud

import (
	"fmt"
	"backend/db"
)

// Create Category
func CreateCategory(name string, description string) {
	category := db.Category{Name: name, Description: description}
	result := db.DB.Create(&category)
	if result.Error != nil {
		fmt.Println("Error creating category:", result.Error)
		return
	}
	fmt.Println("Category created successfully:", category)
}

// Get All Categories
func GetAllCategories() {
	var categories []db.Category
	result := db.DB.Find(&categories)
	if result.Error != nil {
		fmt.Println("Error fetching categories:", result.Error)
		return
	}
	fmt.Println("Categories:", categories)
}

// Update Category
func UpdateCategory(categoryID uint, newName string, newDescription string) {
	var category db.Category
	result := db.DB.First(&category, categoryID)
	if result.Error != nil {
		fmt.Println("Category not found:", result.Error)
		return
	}

	category.Name = newName
	category.Description = newDescription
	db.DB.Save(&category)
	fmt.Println("Category updated successfully:", category)
}

// Delete Category
func DeleteCategory(categoryID uint) {
	result := db.DB.Delete(&db.Category{}, categoryID)
	if result.Error != nil {
		fmt.Println("Error deleting category:", result.Error)
		return
	}
	fmt.Println("Category deleted successfully.")
}

File: final_project/frontend/src/services/categories.js
import api from "./api";

// Category Services
export const fetchCategories = async () =>
  api.get("/categories").then((res) => res.data);

export const fetchCategoryById = async (id) =>
  api.get(`/categories/${id}`).then((res) => res.data);

export const createCategory = async (categoryData) =>
  api.post("/categories", categoryData).then((res) => res.data);

export const updateCategory = async (id, categoryData) =>
  api.put(`/categories/${id}`, categoryData).then((res) => res.data);

export const deleteCategory = async (id) =>
  api.delete(`/categories/${id}`).then((res) => res.status === 204);
